utils/backup/S3/fo-backup-s3

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+# Copyright (C) Siemens AG 2017
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# version 2 as published by the Free Software Foundation.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+source ${DIR}/before
+set -x
+${DIR}/install
+
+echo Starting backup at `date`
+
+
+
+echo Creating diff of repository at `date`
+ls -RlahGg --time-style='+' /srv/fossology/repository/ | cut -d ' ' -f 2-5 | ${DIR}/venv/bin/aws s3 cp --sse AES256 - "s3://${BUCKET}/${FILENAME}-repo-diff.txt"
```

Comment:
Since you are listing recursively, the `s -RlahGg --time-style='+'` also lists the names of the corresponding folders as the first "word" in a line (for sub directories). If they contain spaces you are splitting them and keeping a part of them.